diesel_infer_schema/src/mysql.rs

```diff
@@ -0,0 +1,55 @@
+use std::error::Error;
+
+use data_structures::*;
+
+pub fn determine_column_type(attr: &ColumnInformation) -> Result<ColumnType, Box<Error>> {
+    let tpe = determine_type_name(&attr.type_name);
+
+    Ok(ColumnType {
+        path: vec!["diesel".into(), "types".into(), capitalize(tpe)],
+        is_array: false,
+        is_nullable: attr.nullable,
+    })
+}
+
+fn determine_type_name(sql_type_name: &str) -> &str {
+    if sql_type_name == "tinyint(1)" {
+        "bool"
+    } else if sql_type_name.starts_with("int") {
+        "integer"
+    } else if let Some(idx) = sql_type_name.find('(') {
```

Comment:
I just assume that the actual format is always `(?P<tpe>\w)+\((?P<len>\d)+\)`, i.e., there is a closing bracket, and `len` is always a non-empty string. Validating that should not be our tasks here though. Let's just deal with what MySQL gives us. (So, this is good as-is.)